src/libraries/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.cs

```diff
@@ -1161,7 +1161,7 @@ private bool CheckLiteral(ReadOnlySpan<byte> span, ReadOnlySpan<byte> literal)
                     if (span[i] != literal[i])
                     {
                         _bytePositionInLine += i;
-                        ThrowInvalidLiteral(span);
+                        throw GetInvalidLiteral(span);
```

Comment:
The point of throwing helpers is to minimize the number of instructions being generated in a given method. What has prompted you to change it in this case?

Reply:
The throw helper here was actually being inlined because it was too big for the JIT to immediately see it as non-returning, thus generating a lot code. I could make it into proper throw helper also (that calls `GetInvalidLiteral`), but the additional code size diff would be tiny (only `call CORINFO_HELP_THROW` is saved by that).

Reply:
> I could make it into proper throw helper also

Yes, it would be better to follow uniform throw helper pattern that does not leak the implementation details of the throw helper its callsites.

Many throw helpers are structured like that. For example, https://github.com/dotnet/runtime/blob/d94ebe0770a1ec8d6c554621d19277663dd6d53c/src/libraries/System.Text.Json/src/System/ThrowHelper.cs#L27